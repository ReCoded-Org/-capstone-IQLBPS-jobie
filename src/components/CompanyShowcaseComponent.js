function CompanyShowcaseComponent() {
    return (
      <div className="grid bg-gray-500/5 grid-rows-3 gap-y-5" > 
       
       <div className="justify-self-center self-end w-4/6"> 
         <h2 className="font-inter text-blue-900 font-semibold">More than 20,000 companies are using our service to hire their employees.</h2>
        </div>

          <div className=" grid grid-cols-5 gap-x-5 content-center items-center justify-self-center w-5/6">
            <a href="https://www.facebook.com" rel="noreferrer" target="_blank"> <img className="w-30" src="/assets/company-showcase-component-assets/facebook.png" alt="Facebook's logo"/> </a>
            <a href="https://www.amazon.com" rel="noreferrer" target="_blank"><img className="w-30" src="/assets/company-showcase-component-assets/amazon.png" alt="Amazon's logo"/></a>
            <a href="https://www.microsoft.com" rel="noreferrer" target="_blank"><img className="w-30" src="/assets/company-showcase-component-assets/microsoft.png" alt="Microsoft's logo"/></a>
            <a href="https://www.apple.com" rel="noreferrer" target="_blank"> <img className="w-30" src="/assets/company-showcase-component-assets/apple.png" alt="Apple's logo"/> </a>
            <a href="https://www.google.com" rel="noreferrer" target="_blank"> <img className="w-30 " src="/assets/company-showcase-component-assets/google.png" alt="Google's logo"/></a>
            </div>

          <div className="grid items-center grid-cols-4 gap-x-5 content-center items-center justify-self-center w-5/6">
            <a href="https://www.airbnb.com" rel="noreferrer" target="_blank"><img className = "w-50" src="/assets/company-showcase-component-assets/airbnb.png" alt="Airbnb's logo"/></a>
            <a href="https://www.spotify.com" rel="noreferrer" target="_blank"><img className = "w-50" src="/assets/company-showcase-component-assets/spotify.png" alt="Spotify's logo"/></a>
            <a href="https://www.yahoo.com" rel="noreferrer" target="_blank"><img className = "w-50" src="/assets/company-showcase-component-assets/yahoo.png" alt="Yahoo's logo"/></a>
            <a href="https://www.netflix.com" rel="noreferrer" target="_blank"><img className = "w-50" src="/assets/company-showcase-component-assets/netflix.png" alt="Netflix's logo"/></a>
          </div>

      </div>
    );
  }
  
  export default CompanyShowcaseComponent;