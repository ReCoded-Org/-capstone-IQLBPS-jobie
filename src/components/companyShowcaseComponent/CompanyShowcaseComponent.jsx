function CompanyShowcaseComponent({ showCaseData }) {
  const topRowComanpies = showCaseData.companies.slice(0, 5).map((company) => {
    return (
      <a
        key={`${company}`}
        className="w-50"
        rel="noreferrer"
        target="_blank"
        href={`https://www.${company}.com`}
      >
        <img
          className="hover:scale-75"
          src={`/assets/company-showcase-component-assets/${company}.png`}
          alt={`${company}s logo`}
        />
      </a>
    );
  });

  const bottomRowComanpies = showCaseData.companies
    .slice(5, 10)
    .map((company) => {
      return (
        <a
          key={`${company}`}
          className="w-50"
          rel="noreferrer"
          target="_blank"
          href={`https://www.${company}.com`}
        >
          <img
            className="hover:scale-75"
            src={`/assets/company-showcase-component-assets/${company}.png`}
            alt={`${company}s logo`}
          />
        </a>
      );
    });

  return (
    <section className="flex flex-col bg-gray-500/5 justify-center items-center">
      <div className="grid grid-cols-5 grid-rows-2 gap-x-5 place-content-center items-center w-4/6">
        <h2 className="font-inter text-blue-900 font-semibold self-end text-xs mt-5 mb-5 md:text-2xl col-span-5">
          {showCaseData.header}
        </h2>
        {topRowComanpies}
      </div>
      <div className="grid grid-cols-4 gap-x-5 place-content-center items-center w-5/6">
        {bottomRowComanpies}
      </div>
    </section>
  );
}

export default CompanyShowcaseComponent;
