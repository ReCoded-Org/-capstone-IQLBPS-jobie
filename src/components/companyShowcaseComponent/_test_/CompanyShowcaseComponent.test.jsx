import renderer from 'react-test-renderer';
import CompanyShowcaseComponent from "../CompanyShowcaseComponent.jsx"

it('renders Company Showcase Component correctly', () => {
    const tree = renderer
      .create(<CompanyShowcaseComponent/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

