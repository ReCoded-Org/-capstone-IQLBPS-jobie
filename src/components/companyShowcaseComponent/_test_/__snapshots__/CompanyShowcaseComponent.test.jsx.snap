// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Company Showcase Component correctly 1`] = `
<div
  className="grid bg-gray-500/5 grid-rows-3 gap-y-5"
>
  <div
    className="justify-self-start self-end w-4/6"
  >
    <h2
      className="font-inter text-blue-900 font-semibold"
    >
      More than 20,000 companies are using our service to hire their employees.
    </h2>
  </div>
  <div
    className=" grid grid-cols-5 gap-x-5 content-center items-center justify-self-center w-5/6"
  >
    <a
      href="https://www.facebook.com"
      rel="noreferrer"
      target="_blank"
    >
       
      <img
        alt="Facebook's logo"
        className="w-30"
        src="/assets/company-showcase-component-assets/facebook.png"
      />
       
    </a>
    <a
      href="https://www.amazon.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Amazon's logo"
        className="w-30"
        src="/assets/company-showcase-component-assets/amazon.png"
      />
    </a>
    <a
      href="https://www.microsoft.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Microsoft's logo"
        className="w-30"
        src="/assets/company-showcase-component-assets/microsoft.png"
      />
    </a>
    <a
      href="https://www.apple.com"
      rel="noreferrer"
      target="_blank"
    >
       
      <img
        alt="Apple's logo"
        className="w-30"
        src="/assets/company-showcase-component-assets/apple.png"
      />
       
    </a>
    <a
      href="https://www.google.com"
      rel="noreferrer"
      target="_blank"
    >
       
      <img
        alt="Google's logo"
        className="w-30 "
        src="/assets/company-showcase-component-assets/google.png"
      />
    </a>
  </div>
  <div
    className="grid items-center grid-cols-4 gap-x-5 content-center items-center justify-self-center w-5/6"
  >
    <a
      href="https://www.airbnb.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Airbnb's logo"
        className="w-50"
        src="/assets/company-showcase-component-assets/airbnb.png"
      />
    </a>
    <a
      href="https://www.spotify.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Spotify's logo"
        className="w-50"
        src="/assets/company-showcase-component-assets/spotify.png"
      />
    </a>
    <a
      href="https://www.yahoo.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Yahoo's logo"
        className="w-50"
        src="/assets/company-showcase-component-assets/yahoo.png"
      />
    </a>
    <a
      href="https://www.netflix.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Netflix's logo"
        className="w-50"
        src="/assets/company-showcase-component-assets/netflix.png"
      />
    </a>
  </div>
</div>
`;
